ngraph/core/reference/include/ngraph/runtime/reference/gather_elements.hpp

```diff
@@ -49,7 +49,7 @@ namespace ngraph
                 {
                     for (int64_t i = 0; i < indices_shape[0]; i++)
                     {
-                        if (indices[i] > data_shape[0])
+                        if (indices[i] > data_shape[0] - 1)
```

Comment:
If the `data_shape[0]` is 0 then `size_t(0 - 1) = 2^63 - 1`. Change this line to:
`if (indices[i] + 1 > data_shape[0])`

Reply:
Agree that the current check is not ideal. Your check is better, but the best is to do ` if (indices[i] >= data_shape[0])`. Corrected in other places. Please review.